ApollosProject/apollos-prototype
# Add action list feed feature for upcoming events

## DESCRIPTION

### What does this PR do, or why is it needed?

- Adds the ability to show upcoming events inside the `ActionListFeature`
- Adds a method to fetch "upcoming public" events, or "all events" + a warning if you're not using the Apollos plugin. 
- Adds a method to fetch an image for a given event. 

### What design trade-offs/decisions were made?

Will result in a few additional issues being created

- Need to implement "READ_EVENT" action.
- Related #1029 

Chose not to go overboard making the "UPCOMING_EVENT" algorithm hyper customizable. We can take a look at enhanced functionality in future PRs. 

### How do I test this PR?

1. Open the home feed, you should see an event listed in the "This week at Apollos" section.
2. Look at my pretty picture below of a home feed configured just to show a single upcoming event

<img width="545" alt="Screen Shot 2019-09-16 at 3 40 01 PM" src="https://user-images.githubusercontent.com/1637694/65043289-a152fb00-d928-11e9-8afc-ed49c0fc2864.png">


### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
